homeassistant/components/homewizard_energy/sensor.py

```diff
@@ -0,0 +1,226 @@
+"""Creates Homewizard Energy sensor entities."""
+from __future__ import annotations
+
+import logging
+from typing import Final
+
+from homeassistant.components.sensor import (
+    STATE_CLASS_MEASUREMENT,
+    STATE_CLASS_TOTAL_INCREASING,
+    SensorEntity,
+    SensorEntityDescription,
+)
+from homeassistant.const import (
+    CONF_ID,
+    DEVICE_CLASS_ENERGY,
+    DEVICE_CLASS_GAS,
+    DEVICE_CLASS_POWER,
+    DEVICE_CLASS_TIMESTAMP,
+    ENERGY_KILO_WATT_HOUR,
+    ENTITY_CATEGORY_DIAGNOSTIC,
+    PERCENTAGE,
+    POWER_WATT,
+    VOLUME_CUBIC_METERS,
+)
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import (
+    ATTR_ACTIVE_POWER_L1_W,
+    ATTR_ACTIVE_POWER_L2_W,
+    ATTR_ACTIVE_POWER_L3_W,
+    ATTR_ACTIVE_POWER_W,
+    ATTR_GAS_TIMESTAMP,
+    ATTR_METER_MODEL,
+    ATTR_SMR_VERSION,
+    ATTR_TOTAL_ENERGY_EXPORT_T1_KWH,
+    ATTR_TOTAL_ENERGY_EXPORT_T2_KWH,
+    ATTR_TOTAL_ENERGY_IMPORT_T1_KWH,
+    ATTR_TOTAL_ENERGY_IMPORT_T2_KWH,
+    ATTR_TOTAL_GAS_M3,
+    ATTR_WIFI_SSID,
+    ATTR_WIFI_STRENGTH,
+    CONF_API,
+    CONF_DATA,
+    CONF_MODEL,
+    CONF_SW_VERSION,
+    COORDINATOR,
+    DOMAIN,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+SENSORS: Final[tuple[SensorEntityDescription, ...]] = (
+    SensorEntityDescription(
+        key=ATTR_SMR_VERSION,
+        name="SMR Version",
+        icon="mdi:counter",
+        entity_category=ENTITY_CATEGORY_DIAGNOSTIC,
+    ),
+    SensorEntityDescription(
+        key=ATTR_METER_MODEL,
+        name="Model",
+        icon="mdi:gauge",
+        entity_category=ENTITY_CATEGORY_DIAGNOSTIC,
+    ),
+    SensorEntityDescription(
+        key=ATTR_WIFI_SSID,
+        name="Wifi SSID",
+        icon="mdi:wifi",
+        entity_category=ENTITY_CATEGORY_DIAGNOSTIC,
+    ),
+    SensorEntityDescription(
+        key=ATTR_WIFI_STRENGTH,
```

Comment:
Disable this one by default. It spams.

Reply:
https://github.com/DCSBL/core/commit/2b74b7715dd2e2d630a1f2bba1d7fdd24c195c7f